easypost/__init__.py

```diff
@@ -154,90 +153,45 @@ def convert_to_easypost_object(response, api_key, parent=None, name=None):
-        if isinstance(value, six.text_type):
-            return value.encode('utf-8')
-        return value
-    elif isinstance(value, six.binary_type):
-        # Python3's six.text_type(bytestring) returns "b'bytestring'"
-        # So, have to decode it to unicode
+    # Python3's str(bytestring) returns "b'bytestring'" so we use .decode instead
+    if isinstance(value, six.binary_type):
         return value.decode('utf-8')
-    else:
-        # Python3's urlencode can handle unicode
-        return value
+    return value
 
 
-class Requestor(object):
-    def __init__(self, local_api_key=None):
-        self._api_key = local_api_key
+def encode_url_params(params):
+    converted_params = []
+    for key, value in sorted(six.iteritems(params)):
+        if value is None:
+            continue  # don't add Nones to the query
+        elif isinstance(value, datetime.datetime):
+            value = int(time.mktime(value.timetuple()))  # to UTC timestamp
+        converted_params.append((key, value))
+    return urlencode(converted_params)
 
-    @classmethod
-    def api_url(cls, url=None):
-        url = url or ''
-        return '%s%s' % (api_base, url)
 
-    @classmethod
-    def encode_datetime(cls, out, key, dt_value):
-        utc_timestamp = int(time.mktime(dt_value.timetuple()))
-        out.append((key, utc_timestamp))
+def add_params_to_url(url, params):
+    encoded_params = encode_url_params(params)
+    if encoded_params:
+        return '%s?%s' % (url, encoded_params)
+    return url
 
-    @classmethod
-    def encode_none(cls, out, key, value):
-        pass  # do not include None-valued params in request
 
-    @classmethod
-    def encode_url(cls, params):
-        # special case value encoding
-        ENCODERS = {
-            datetime.datetime: cls.encode_datetime,
-        }
-        if six.PY2:
-            ENCODERS[types.NoneType] = cls.encode_none
-        if six.PY3:
-            ENCODERS[type(None)] = cls.encode_none
-
-        out = []
-        for key, value in sorted(six.iteritems(params)):
-            key = _utf8(key)
-            try:
-                encoder = ENCODERS[value.__class__]
-                encoder(out, key, value)
-            except KeyError:
-                # don't need special encoding
-                try:
-                    value = six.text_type(value)
-                except Exception:
-                    pass
-
-                out.append((key, value))
-        return urlencode(out)
+class Requestor(object):
+    def __init__(self, local_api_key=None):
+        self._api_key = local_api_key
 
     @classmethod
     def _objects_to_ids(cls, param):
         if isinstance(param, Resource):
             return {'id': param.id}
         elif isinstance(param, dict):
-            out = {}
-            for k, v in six.iteritems(param):
-                out[k] = cls._objects_to_ids(v)
-            return out
+            return {k: cls._objects_to_ids(v) for k, v in six.iteritems(param)}
```

Comment:
It could be advantageous to name these return values for easy code scanning?

Reply:
I'm very amenable if anyone has a name that isn't just a name for its own sake.
`{k: f(v) for k, v in dict.items()}` and `[f(v) for v in list]` are such common and simple idioms that a name would have to be very expressive to improve clarity here.

```
param_with_Resources_as_their_ids = {k: cls._objects_to_ids(v) for k, v in six.iteritems(param)}
return param_with_Resources_as_their_ids
```
? 😄